sklearn/utils/estimator_checks.py

```diff
@@ -67,7 +67,14 @@
                 'RandomForestRegressor', 'Ridge', 'RidgeCV']
 
 
-def _yield_non_meta_checks(name, Estimator):
+def assert_almost_equal_dense_sparse(x, y, err_msg=''):
+    if sparse.issparse(x):
+        assert_allclose(x.data, y.data, err_msg=err_msg)
```

Comment:
I think this helper is wrong in many cases. For instance:

```
from scipy import sparse as sp
from sklearn.utils.testing import assert_allclose
import numpy as np

A = sp.diags(np.ones(5)).tocsr()
B = sp.csr_matrix(np.ones((1, 5)))

assert_allclose(B.data, A.data)
```

Comparing data only makes sense if x and y are the same sparse format, same shape, and same sparsity pattern, as far as I can tell.